src/librustc_metadata/locator.rs

```diff
@@ -856,6 +858,19 @@ fn get_metadata_section(target: &Target,
     return ret;
 }
 
+/// A trivial wrapper for `Mmap` that implements `StableDeref`.
+struct StableDerefMmap(memmap::Mmap);
+
+impl Deref for StableDerefMmap {
+    type Target = [u8];
+
+    fn deref(&self) -> &[u8] {
+        self.0.deref()
```

Comment:
My preferred way to write this is `&*self.0`, but YMMV.

Reply:
In general, I agree, but I think this way is clearer within a `deref` method :)